File: wings-silencer/src/test/java/pro/fessional/wings/silencer/i18n/MessagePrintTest.java
package pro.fessional.wings.silencer.i18n;

import lombok.Setter;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.MessageSource;
import pro.fessional.mirana.data.Arr;

import java.util.Locale;

import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * @author trydofor
 * @since 2019-06-10
 */
@SpringBootTest
public class MessagePrintTest {

    @Setter(onMethod_ = {@Autowired})
    private MessageSource messageSource;

    @Setter(onMethod_ = {@Value("${wings.test.module}")})
    private String module;

    @Test
    void print() {
        String cn = messageSource.getMessage("base.not-empty", Arr.of("姓名"), Locale.CHINA);
        String en = messageSource.getMessage("base.not-empty", Arr.of("name"), Locale.US);
        assertEquals("姓名 不能为空", cn);
        assertEquals("name can not be empty", en);
        assertEquals("沉默术士", module);
    }
}

File: wings-faceless-flywave/src/test/kotlin/pro/fessional/wings/faceless/i18n/MessagePrintTest.kt
package pro.fessional.wings.faceless.i18n

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.MessageSource
import java.util.Locale

/**
 * @author trydofor
 * @since 2019-06-10
 */
@SpringBootTest
class MessagePrintTest {

    @Autowired
    lateinit var messageSource: MessageSource

    // 真难看的\$
    @Value("\${wings.test.module}")
    lateinit var module: String

    @Test
    fun print() {
        val cn = messageSource.getMessage("base.not-empty", arrayOf("姓名"), Locale.CHINA)
        val en = messageSource.getMessage("base.not-empty", arrayOf("name"), Locale.US)
        assertEquals("姓名 不能为空", cn)
        assertEquals("name can not be empty", en)
        assertEquals("虚空假面", module)
    }
}